recipe/meta.yaml

```diff
@@ -21,18 +22,24 @@ requirements:
     - python >=3.7
     - geopandas-base >=0.10
     - dask-core >=2021.06.0
-    - pygeos
+    - distributed >=2021.06.0
```

Comment:
I am also inclined to remove this here, given that we explicitly depend on "dask-core" and not "dask", exactly to avoid pulling in all dependencies of dask in case you don't need them

Reply:
But distributed is defined as required in https://github.com/geopandas/dask-geopandas/blob/v0.3.1/setup.py#L16? I actually got an error that distributed was missing when running `pip check`, which was why I added it.

Reply:
Yeah, that's true. I don't really remember the discussions about adding that there and not here. Now, pypi dependencies don't have this concept of dask-core and geopandas-base. 

It might have been added for the pypi package because if you do `pip install dask` you do _not_ get distributed with it, while if you do `conda install dask` you do already get distributed as well.

Reply:
Ok, removed `distributed`, let's see if the checks pass.

Reply:
Yep, getting an error without `distributed` at https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=762329&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=986b1512-c876-5f92-0d81-ba851554a0a3&l=598:

```
export PREFIX=/home/conda/feedstock_root/build_artifacts/dask-geopandas_1692098287394/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p
export SRC_DIR=/home/conda/feedstock_root/build_artifacts/dask-geopandas_1692098287394/test_tmp
import: 'dask_geopandas'
import: 'dask_geopandas'
+ pip check
dask-geopandas 0.3.1 requires distributed, which is not installed.
Tests failed for dask-geopandas-0.3.1-pyhd8ed1ab_1.conda - moving package to /home/conda/feedstock_root/build_artifacts/broken
```

We could possibly patch things out here to get `pip check` to work, but ideally, the dependency specification should be fixed upstream at https://github.com/geopandas/dask-geopandas, and the conda-forge package would just follow that.

Reply:
It looks like setup.py has always specified that distributed is required. So I think it is more correct to include distributed in the conda-forge recipe as well. 

It would probably be better if dask-geopandas did not depend on distributed but that feels like a separate discussion.

Reply:
Ok, added `distributed` again in 7af3f6f457aba19eee3d7d85f7273a1def60c8ca. I've opened up an issue at https://github.com/geopandas/dask-geopandas/issues/258 to continue the `distributed` as a required dependency discussion.

Reply:
It's expected that the pip check will fail, because the main thing it does is compare those dependencies, and as we have been discussing, the pip dependencies currently include `distributed`. 

To avoid dragging this PR on the distributed discussion, I pushed a small change to remove both distributed and the pip check, so we keep this PR solely focused on removing the pygeos dependency, which is useful to get in. I hope you don't mind, and thanks for the work here! It will be good to sort out the dependency discussion and make it more consistent between pip and conda.